### PR TITLE
Add getting started and installation instructions to the welcome page

### PR DESCRIPTION
**Pull request by weiji14**
Editing the front page to include a one-click Binder link to run the Jupyter notebooks, and steps for people to get a conda virtual environment set up locally for the AGU24 pre-conference workshop.

**Screenshot**

image

TODO:
- [x] Adding quickstart binder button and installation instructions
- [x] Add sphinx-design cards in 3x2 grid layout for the 6 tutorials
- [x] Include thumbnails of the best looking figure in each tutorial (do after all tutorials are ready)
- [ ] etc

References:
- Based on https://github.com/GenericMappingTools/egu22pygmt/pull/10
- Sphinx design cards - https://sphinx-design.readthedocs.io/en/sbt-theme/cards.html

### CODE REVIEW

File: book/intro.md [hunk 1]
```diff
@@ -1,11 +1,158 @@
...
+by [Yvonne Fröhlich](https://orcid.org/0000-0002-8566-0619)
++++
+{bdg-primary}`pygmt`
+{bdg-primary-line}`histogram`
+{bdg-primary-line}`legend`
+{bdg-primary-line}`plot`
+{bdg-secondary-line}`japan_quakes`
+{bdg-info}`pandas`
+{bdg-info}`geopandas`
+{bdg-info-line}`airbnb`
```
Comment (yvonnefroehlich) [inline book/intro.md]: Not sure about this; I would prefer to include "choropleth map"
```suggestion
{bdg-info-line}`choropleth map`
```

Comment (weiji14) [reply book/intro.md]: Yeah, I could use some advice on what tags/badges to add. My current thought process is to include:

- bdg-primary: pygmt or gmt
- bdg-primary-line: GMT module used (e.g. `coast`, `plot`, etc)
- bdg-info: External library used (e.g. `pandas`, `xarray`)
- bdg-secondary-line: GMT remote-dataset used (e.g. `earth_relief`, `earth_age`, etc)

The ``` {bdg-info-line}`airbnb` ``` here was intended to show external non GMT remote-datasets used, but I do agree that having AirBnB here looks weird. We could maybe add some tags for concepts being taught such as `choropleth-map` as you suggested, or `scatterplot`, etc.

Ref: color of badges/tags: https://sphinx-design.readthedocs.io/en/sbt-theme/badges_buttons.html#badges

**Review by yvonnefroehlich — COMMENTED** [review]
~So far, I did not figure out how to make a useful code suggestion in case an image is included in this cell of the Jupyter notebook. Thus I just list these two aspects:~

I am very sorry I posted this on the wrong PR, and in this PR I have no permission to delete a review or comment.

File: book/intro.md (hunk 1 above)
Comment (yvonnefroehlich) [reply book/intro.md]: Yep, I got your main idea behind the structure of these tags / badges and I definitely like this idea!  And I think mentioning special concept / specific visualization is a good addition.  

Even the file contains "airbnb" currently the column “population” is used for the choropleth-map (there are a lot of different columns / quantities in this dataset).

**Comment by weiji14** [comment]
Ok, spent way too much time on the grid layout, but this PR is finally ready for review! Screenshot of the thumbnail images on the front page:

image

Had to do `mamba install 'pydata-sphinx-theme=0.15.3'` to get the left sidebar to hide away properly (xref https://github.com/jupyter-book/jupyter-book/issues/2160). I might pin that dependency later, but otherwise, please review this and I'll try to merge this tomorrow.

File: book/intro.md [hunk 2]
```diff
@@ -1,11 +1,207 @@
...
+{bdg-primary-line}`coast`
+{bdg-primary-line}`colorbar`
+{bdg-primary-line}`grdimage`
+{bdg-primary-line}`makecpt`
+{bdg-primary-line}`subplot`
+{bdg-secondary-line}`earth_relief`
+{bdg-secondary-line}`earth_age`
+:::
+
+:::{grid-item-card} Tutorial 2 - Integration with the Scientific Python Ecosystem: Pandas / GeoPandas
```
Comment (yvonnefroehlich) [inline book/intro.md]:
```suggestion
:::{grid-item-card} Tutorial 2 - Integration with the scientific Python ecosystem: pandas / GeoPandas
```

File: book/intro.md [hunk 3]
```diff
@@ -1,11 +1,207 @@
...
+
+::::
+:::::
+
+
+:::::{grid-item}
+::::{grid} 1 1 1 1
+:gutter: 1
+
+:::{grid-item-card} Tutorial 3 - Integration with the Scientific Python Ecosystem: Xarray (grids)
```
Comment (yvonnefroehlich) [inline book/intro.md]:
```suggestion
:::{grid-item-card} Tutorial 3 - Integration with the scientific Python ecosystem: Xarray (grids)
```

**Review by yvonnefroehlich — COMMENTED** [review]
Just a few typo fixes for now - but this looks already very nice @weiji14 🙂! Will give this PR a more detailed review later the day!

File: book/intro.md (hunk 2 above)
Comment (jhtong33) [reply book/intro.md]: I guess @weiji14 try to maintain consistency across all titles by capitalizing the first letter of each word throughout the page. Maybe it's just to ensure consistency on this page or to align with the style used in EGU22?

**Review by yvonnefroehlich — COMMENTED** [review]
Thanks, @weiji14, for working on the introduction page. This looks nice 🙂!

I personally like equal-sized cards more, at least a common horizontal line at which the upper edges of the cards of one row align. But if this is difficult to set up, we can leave the layout as is. Also for the EGU workshop, the height of the cards scales with the size of the images (please see https://www.generic-mapping-tools.org/egu22pygmt/intro.html).

File: book/intro.md [hunk 4]
```diff
@@ -1,11 +1,207 @@
...
++++
+{bdg-primary}`pygmt`
+{bdg-primary-line}`grdcontour`
+{bdg-primary-line}`grdinfo`
+{bdg-primary-line}`grdtrack`
+{bdg-primary-line}`meca`
+{bdg-primary-line}`project`
+{bdg-primary-line}`text`
+{bdg-primary-line}`xyz2grd`
+{bdg-success}`numpy`
```
Comment (yvonnefroehlich) [inline book/intro.md]:
```suggestion
{bdg-primary-line}`xyz2grd`
{bdg-secondary-line}`earth_relief`
{bdg-success}`numpy`
```

File: book/intro.md [hunk 5]
```diff
@@ -1,11 +1,207 @@
...
++++
+{bdg-primary}`pygmt`
+{bdg-primary-line}`histogram`
+{bdg-primary-line}`legend`
+{bdg-primary-line}`plot`
+{bdg-secondary-line}`japan_quakes`
+{bdg-success}`pandas`
+{bdg-success}`geopandas`
+{bdg-success-line}`choropleth map`
+:::
```
Comment (yvonnefroehlich) [inline book/intro.md]:
```suggestion
{bdg-success}`geopandas`
{bdg-success-line}`choropleth map`
{bdg-success-line}`scatter plot`
:::
```

File: book/intro.md (hunk 2 above)
Comment (weiji14) [reply book/intro.md]: I'll take @yvonnefroehlich's suggestions on capitalization, we should try to match with the titles on the left side bar too.

**Comment by weiji14** [comment]
> Thanks, @weiji14, for working on the introduction page. This looks nice 🙂!
> 
> I personally like equal-sized cards more, at least a common horizontal line at which the upper edges of the cards of one row align. But if this is difficult to set up, we can leave the layout as is. Also for the EGU workshop, the height of the cards scales with the size of the images (please see https://www.generic-mapping-tools.org/egu22pygmt/intro.html).

Yeah, so the previous EGU22 short course used `sphinx-panels` which is now deprecated in favour of `sphinx-design` (see https://sphinx-design.readthedocs.io/en/sbt-theme/get_started.html#migrating-from-sphinx-panels), and I couldn't quite work out how to get 2 rows with a range of different heights to look nice (the EGU22 one only had 1 rows which was easier). I did try with equal-height cards per-row, but because each thumbnail figure has very different heights (e.g. Tutorial 4 had an extremely tall figure, compared to Tutorial 5 which is very short), there would be a lot of whitespace that didn't look very nice, so I've adapted the grid layout in commit ecf863f69a57e9b9afa5736bcd7b153a2cb068a7. 

I'll merge this in first, so we at least have the installation instructions on the website (and @maxrjones can send the email to the participants). If anyone has time to alter the thumbnail layout, feel free to do that (but I'm guessing there are more important things to fix before that)!

**Comment by yvonnefroehlich** [comment]
> > I personally like equal-sized cards more, at least a common horizontal line at which the upper edges of the cards of one row align. But if this is difficult to set up, we can leave the layout as is. Also for the EGU workshop, the height of the cards scales with the size of the images (please see https://www.generic-mapping-tools.org/egu22pygmt/intro.html).
> 
> Yeah, so the previous EGU22 short course used `sphinx-panels` which is now deprecated in favour of `sphinx-design` (see https://sphinx-design.readthedocs.io/en/sbt-theme/get_started.html#migrating-from-sphinx-panels), and I couldn't quite work out how to get 2 rows with a range of different heights to look nice (the EGU22 one only had 1 rows which was easier). I did try with equal-height cards per-row, but because each thumbnail figure has very different heights (e.g. Tutorial 4 had an extremely tall figure, compared to Tutorial 5 which is very short), there would be a lot of whitespace that didn't look very nice, so I've adapted the grid layout in commit [ecf863f](https://github.com/GenericMappingTools/agu24workshop/commit/ecf863f69a57e9b9afa5736bcd7b153a2cb068a7).

No worries. I already thought based on the comment "spent way too much time on the grid layout" that you already tried different layouts 🙈.
If people really want to work with the materials and play around with the codes, they will download the JNs. So, I do not think the layout of the cards is too important, and we should focus on more relevant aspects first.

**Pull request merged**